test/utils/image/manifest.go

```diff
@@ -358,8 +347,15 @@ func GetPauseImageName() string {
 	return GetE2EImage(Pause)
 }
 
-// ReplaceRegistryInImageURL replaces the registry in the image URL with a custom one
+// ReplaceRegistryInImageURL replaces the registry in the image URL with a custom one based
+// on the configured registries.
 func ReplaceRegistryInImageURL(imageURL string) (string, error) {
+	return replaceRegistryInImageURLWithList(imageURL, registry)
```

Comment:
Exported method uses the package level variable; the logic was extracted into an unexported method where we can pass the RegistryList for easier testing/injection.